packages/nuxt/src/imports/module.ts

```diff
@@ -117,7 +117,25 @@ export default defineNuxtModule<Partial<ImportsOptions>>({
           const scannedImports = await scanDirExports(composablesDirs, {
             fileFilter: file => !isIgnored(file),
           })
+
+          const presetMap = new Map<string, string>()
+
+          for (const preset of presets) {
+            preset.imports = preset.imports ?? []
+            for (const i of preset.imports) {
+              presetMap.set(i, preset.from)
+            }
+            presetMap.set(preset.as, preset.from)
```

Comment:
the previous implementation has type issues. types are weired on the presets. I tried to solve it like this, but seems kinda hacky
```suggestion
            for (const i of preset.imports) {
              if(typeof i === "string") {
                presetMap.set(i, preset.from)
                continue
              }
              let name, as, from
              if(Array.isArray(i)) {
                [name, as, from] = i
              }
              else {
                name = i?.name ?? undefined;
                as = i?.as ?? undefined;
                from = preset.from
              }
              if(!name || !from) {
                continue
              }
              presetMap.set(as ?? name, from)
            }
```